File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/novelextractor/NovelVolume.java
package org.jisho.textosJapones.model.entities.novelextractor;

import com.google.gson.annotations.Expose;
import org.jisho.textosJapones.model.enums.Language;

import java.util.*;

public class NovelVolume {

    private UUID id;
    @Expose
    private String novel;
    @Expose
    private String titulo;
    @Expose
    private String tituloAlternativo;
    @Expose
    private String serie;
    @Expose
    private String descricao;
    @Expose
    private String arquivo;
    @Expose
    private String editora;
    @Expose
    private String autor;
    @Expose
    private Float volume;
    @Expose
    private Language lingua;

    private NovelCapa capa;
    private Boolean favorito;
    @Expose
    private List<NovelCapitulo> capitulos;
    @Expose
    private Set<NovelVocabulario> vocabularios;
    private Boolean processado;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getNovel() {
        return novel;
    }

    public void setNovel(String novel) {
        this.novel = novel;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getTituloAlternativo() {
        return tituloAlternativo;
    }

    public void setTituloAlternativo(String tituloAlternativo) {
        this.tituloAlternativo = tituloAlternativo;
    }

    public String getSerie() {
        return serie;
    }

    public void setSerie(String serie) {
        this.serie = serie;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public String getArquivo() {
        return arquivo;
    }

    public void setArquivo(String arquivo) {
        this.arquivo = arquivo;
    }

    public String getEditora() {
        return editora;
    }

    public void setEditora(String editora) {
        this.editora = editora;
    }

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public Float getVolume() {
        return volume;
    }

    public void setVolume(Float volume) {
        this.volume = volume;
    }

    public Language getLingua() {
        return lingua;
    }

    public void setLingua(Language lingua) {
        this.lingua = lingua;
    }

    public void setFavorito(Boolean favorito) {
        this.favorito = favorito;
    }

    public Boolean isFavorito() {
        return favorito;
    }

    public NovelCapa getCapa() {
        return capa;
    }

    public void setCapa(NovelCapa capa) {
        this.capa = capa;
    }

    public List<NovelCapitulo> getCapitulos() {
        return capitulos;
    }

    public void setCapitulos(List<NovelCapitulo> capitulos) {
        this.capitulos = capitulos;
    }

    public void addCapitulos(NovelCapitulo capitulo) {
        this.capitulos.add(capitulo);
    }

    public Set<NovelVocabulario> getVocabularios() {
        return vocabularios;
    }

    public void setVocabularios(Set<NovelVocabulario> vocabularios) {
        this.vocabularios = vocabularios;
    }

    public void addVocabulario(NovelVocabulario vocabulario) {
        this.vocabularios.add(vocabulario);
    }

    public Boolean getProcessado() {
        return processado;
    }

    public void setProcessado(Boolean processado) {
        this.processado = processado;
    }

    public NovelVolume() {
        this.id = null;
        this.novel = "";
        this.titulo = "";
        this.tituloAlternativo = "";
        this.serie = "";
        this.descricao = "";
        this.editora = "";
        this.autor = "";
        this.volume = 0f;
        this.lingua = Language.PORTUGUESE;
        this.arquivo = "";
        this.processado = false;
        this.favorito = false;
        this.capa = null;
        this.vocabularios = new HashSet<>();
        this.capitulos = new ArrayList<>();
    }

    public NovelVolume(UUID id, String novel, String titulo, String tituloAlternativo, String serie, String descricao, String arquivo, String editora, String autor, Float volume, Language lingua, Boolean favorito, Boolean processado) {
        this.id = id;
        this.novel = novel;
        this.titulo = titulo;
        this.tituloAlternativo = tituloAlternativo;
        this.serie = serie;
        this.descricao = descricao;
        this.arquivo = arquivo;
        this.editora = editora;
        this.autor = autor;
        this.volume = volume;
        this.lingua = lingua;
        this.processado = processado;
        this.favorito = favorito;
        this.capa = null;
        this.vocabularios = new HashSet<>();
        this.capitulos = new ArrayList<>();
    }

    public NovelVolume(UUID id, String novel, String titulo, String tituloAlternativo, String serie, String descricao, String arquivo, String editora, String autor, Float volume, Language lingua,
                       Boolean favorito, NovelCapa capa, Boolean processado, List<NovelCapitulo> capitulos, Set<NovelVocabulario> vocabularios) {
        this.id = id;
        this.novel = novel;
        this.titulo = titulo;
        this.tituloAlternativo = tituloAlternativo;
        this.serie = serie;
        this.descricao = descricao;
        this.arquivo = arquivo;
        this.editora = editora;
        this.autor = autor;
        this.volume = volume;
        this.lingua = lingua;
        this.favorito = favorito;
        this.capa = capa;
        this.capitulos = capitulos;
        this.vocabularios = vocabularios;
        this.processado = processado;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        NovelVolume that = (NovelVolume) o;
        return Objects.equals(id, that.id) && Objects.equals(novel, that.novel) && Objects.equals(volume, that.volume) && lingua == that.lingua;
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), id, novel, volume, lingua);
    }

    @Override
    public String toString() {
        return "NovelVolume [" + "id=" + id + ", novel=" + novel + ", titulo=" + titulo + ", tituloAlternativo=" + tituloAlternativo + ", descricao=" + descricao +
                ", arquivo=" + arquivo + ", editora=" + editora + ", volume=" + volume + ", lingua=" + lingua + "]";
    }
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/novelextractor/NovelVolume.kt
package org.jisho.textosJapones.model.entities.novelextractor

import br.com.fenix.processatexto.model.entities.Novel
import br.com.fenix.processatexto.model.entities.novelextractor.NovelCapitulo
import br.com.fenix.processatexto.model.entities.processatexto.VocabularioExterno
import br.com.fenix.processatexto.model.enums.Language
import com.google.gson.annotations.Expose
import java.util.*

data class NovelVolume(
    var id: UUID? = null,
    @Expose override var novel: String = "",
    @Expose var titulo: String = "",
    @Expose var tituloAlternativo: String = "",
    @Expose var serie: String = "",
    @Expose var descricao: String = "",
    @Expose var arquivo: String = "",
    @Expose var editora: String = "",
    @Expose var autor: String = "",
    @Expose override var volume: Float = 0f,
    @Expose var lingua: Language = Language.PORTUGUESE,
    var capa: NovelCapa? = null,
    var isFavorito: Boolean = false,
    @Expose var capitulos: MutableList<NovelCapitulo> = mutableListOf(),
    @Expose var vocabularios: MutableSet<VocabularioExterno> = mutableSetOf(),
    var processado: Boolean = false
) : Novel() {


    constructor(
        id: UUID?,
        novel: String,
        titulo: String,
        tituloAlternativo: String,
        serie: String,
        descricao: String,
        arquivo: String,
        editora: String,
        autor: String,
        volume: Float,
        lingua: Language,
        favorito: Boolean,
        processado: Boolean
    ) : this() {
        this.id = id
        this.novel = novel
        this.titulo = titulo
        this.tituloAlternativo = tituloAlternativo
        this.serie = serie
        this.descricao = descricao
        this.arquivo = arquivo
        this.editora = editora
        this.autor = autor
        this.volume = volume
        this.lingua = lingua
        this.processado = processado
        capitulo = 0f
        isFavorito = favorito
        capa = null
        vocabularios = mutableSetOf()
        capitulos = mutableListOf()
    }

    constructor(
        id: UUID?,
        novel: String,
        titulo: String,
        tituloAlternativo: String,
        serie: String,
        descricao: String,
        arquivo: String,
        editora: String,
        autor: String,
        volume: Float,
        lingua: Language,
        favorito: Boolean,
        capa: NovelCapa?,
        processado: Boolean,
        capitulos: MutableList<NovelCapitulo>,
        vocabularios: MutableSet<VocabularioExterno>
    ) : this() {
        this.id = id
        this.novel = novel
        this.titulo = titulo
        this.tituloAlternativo = tituloAlternativo
        this.serie = serie
        this.descricao = descricao
        this.arquivo = arquivo
        this.editora = editora
        this.autor = autor
        this.volume = volume
        this.lingua = lingua
        isFavorito = favorito
        this.capa = capa
        this.capitulos = capitulos
        this.vocabularios = vocabularios
        this.processado = processado
        capitulo = 0f
    }

    fun addCapitulos(capitulo: NovelCapitulo) = capitulos.add(capitulo)

    override fun toString(): String {
        return "NovelVolume [" + "id=" + id + ", novel=" + novel + ", titulo=" + titulo + ", tituloAlternativo=" + tituloAlternativo + ", descricao=" + descricao +
                ", arquivo=" + arquivo + ", editora=" + editora + ", volume=" + volume + ", lingua=" + lingua + "]"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        if (!super.equals(other)) return false

        other as NovelVolume

        if (id != other.id) return false
        if (novel != other.novel) return false
        if (volume != other.volume) return false
        if (lingua != other.lingua) return false

        return true
    }

    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 31 * result + (id?.hashCode() ?: 0)
        result = 31 * result + novel.hashCode()
        result = 31 * result + volume.hashCode()
        result = 31 * result + lingua.hashCode()
        return result
    }
}